# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/media/webcodecs/VideoDecoder.cpp
# Commit: 2e334c8a1160
# Full Hash: 2e334c8a11602d2783fd894a237f170267b1fa94
# Author: Paul Adenot <paul@paul.cx>
# Date: 2023-12-23 09:52:03
# Regressor Bug: 1749047
# File Overlap Count: 3
# Description:
#   Bug 1749047 - Allow printing a VideoDecoderConfig and a VideoDecoderConfigInternal. r=chunmin
#   
#   This is very useful to understand in the log what config is output in what
#   encoder output callback, with respect to configure(...) calls.
#   
# ==============================================================================

diff -r be191f9ac293 -r 2e334c8a1160 dom/media/webcodecs/VideoDecoder.cpp
--- a/dom/media/webcodecs/VideoDecoder.cpp	Fri Dec 22 21:39:13 2023 +0000
+++ b/dom/media/webcodecs/VideoDecoder.cpp	Fri Dec 22 21:39:14 2023 +0000
@@ -154,6 +154,34 @@
       OptionalToMaybe(aConfig.mOptimizeForLatency)));
 }
 
+nsString VideoDecoderConfigInternal::ToString() {
+  nsString rv;
+
+  rv.Append(mCodec);
+  if (mCodedWidth.isSome()) {
+    rv.AppendPrintf("coded: %dx%d", mCodedWidth.value(), mCodedHeight.value());
+  }
+  if (mDisplayAspectWidth.isSome()) {
+    rv.AppendPrintf("display %dx%d", mDisplayAspectWidth.value(),
+                    mDisplayAspectHeight.value());
+  }
+  if (mColorSpace.isSome()) {
+    rv.AppendPrintf("colorspace %s", "todo");
+  }
+  if (mDescription.isSome()) {
+    rv.AppendPrintf("extradata: %zu bytes", mDescription.value()->Length());
+  }
+  rv.AppendPrintf(
+      "hw accel: %s",
+      HardwareAccelerationValues::GetString(mHardwareAcceleration).data());
+  if (mOptimizeForLatency.isSome()) {
+    rv.AppendPrintf("optimize for latency: %s",
+                    mOptimizeForLatency.value() ? "true" : "false");
+  }
+
+  return rv;
+}
+
 /*
  * The followings are helpers for VideoDecoder methods
  */